ReSharper.FSharp/src/FSharp.Psi.Features/src/CodeStyle/FantomasOptionsPage.fs

```diff
@@ -0,0 +1,350 @@
+        delayedNotifications <- []
+
+    let recalculateState versionSetting =
+        let version, warnings = calculateRunSettings versionSetting
+        versionToRun.Value <- version
+        delayedNotifications <- warnings
+
+    let validate version (pathToExecutable: VirtualFileSystemPath) =
+        match Version.TryParse(version) with
+        | true, version when version >= minimalSupportedVersion ->
+            if isNotNull pathToExecutable && pathToExecutable.ExistsDirectory then Ok else FailedToRun
+        | _ -> UnsupportedVersion
+
+    let invalidateDotnetTool toolVersion toolInfo =
+        match toolInfo with
+        | Some (version, pathToExecutable: VirtualFileSystemPath) ->
+            match versionsData.TryGetValue(toolVersion) with
+            | true, { Version = _, cachedVersion } when version = cachedVersion -> ()
+            | _ ->
+                versionsData.Remove(toolVersion) |> ignore
+                notificationsFired.Remove(toolVersion) |> ignore
+                let fantomasDirectory = if isNull pathToExecutable then null else pathToExecutable.Directory
+                let status = validate version fantomasDirectory
+                versionsData.Add(toolVersion, { Version = toolVersion, version; Path = fantomasDirectory; Status = status })
+        | _ ->
+            versionsData.Remove(toolVersion) |> ignore
+            notificationsFired.Remove(toolVersion) |> ignore
+    
+    let invalidateDotnetTools (cache: ToolManifestCache) selectedVersion =
+        cache.LocalToolManifestInfo.Tools
+        |> Seq.tryFind (fun x -> x.Info.PackageId = fantomasPackageId || x.Info.PackageId = fantomasToolPackageId)
+        |> Option.map (fun x -> x.Info.Version, x.PathToExecutable)
+        |> invalidateDotnetTool SolutionDotnetTool
+
+        cache.GlobalToolManifests
+        |> Seq.tryFind (fun x -> (x.Name = fantomasPackageId || x.Name = fantomasToolPackageId) && x.Versions.Count > 0)
+        |> Option.map (fun x ->
+            let version = (Seq.max x.Versions).OriginalVersion
+            version, VirtualFileSystemPath.Parse(calculateGlobalFantomasToolPath version, InteractionContext.Local))
+        |> invalidateDotnetTool GlobalDotnetTool
+
+        match selectedVersion with
+        | FantomasVersionSettings.SolutionDotnetTool ->
+            if versionsData.ContainsKey(SolutionDotnetTool) then () else
+            versionsData.Add(SolutionDotnetTool, { Version = SolutionDotnetTool, ""; Path = null; Status = SelectedButNotFound })
+
+        | FantomasVersionSettings.GlobalDotnetTool ->
+            if versionsData.ContainsKey(GlobalDotnetTool) then () else
+            versionsData.Add(GlobalDotnetTool, { Version = GlobalDotnetTool, ""; Path = null; Status = SelectedButNotFound })
+
+        | _ -> ()
+
+    do
+        if isNull settingsProvider || isNull dotnetToolsTracker then () else
+
+        settingsProvider.Version.Change.Advise(lifetime, fun x ->
```

Comment:
What do you think about using `fantomasSettingsProvider` name? Or `fantomasVersionProvider`? It'd one less thing to guess when looking at this constructor.